### PR TITLE
Issue #591 Serialize SecurityKey to JsonWebKey

### PR DESCRIPTION
**Pull request by lovemaths**

### CODE REVIEW

File: test/Microsoft.IdentityModel.Tokens.Tests/JsonWebKeyConverterTest.cs [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+        public static TheoryData<ConverterTheoryData> JsonWebKeyConverterTestTheoryData
+        {
+            get
+            {
+                return new TheoryData<ConverterTheoryData>()
+                {
+                    new ConverterTheoryData
+                    {
+                        SecurityKey = KeyingMaterial.RsaSecurityKey_2048,
```
Comment (brentschmaltz) [inline test/Microsoft.IdentityModel.Tokens.Tests/JsonWebKeyConverterTest.cs]: Stick a First=true here.

**Comment by brentschmaltz** [comment]
can you add that small change, then commit to dev.
make sure to run all tests before pushing to dev after rebase as we had quite a few changes to IdentityComparer.

**Comment by brentschmaltz** [comment]
@lovemaths @mafurman has committed #684 you can make this change now.

**Comment by lovemaths** [comment]
@brentschmaltz rebased.

File: test/Microsoft.IdentityModel.Tokens.Tests/JsonWebKeyConverterTest.cs [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+            {
+                var jsonWebKey = JsonWebKeyConverter.ConvertFromSecurityKey(theoryData.SecurityKey);
+                if (theoryData.SecurityKey.GetType() == typeof(X509SecurityKey))
+                {
+                    theoryData.ExpectedException.ProcessNoException(context);
+                    IdentityComparer.AreEqual(jsonWebKey.Kty, theoryData.ComparisonJsonWebKey.Kty, context);
+                    IdentityComparer.AreEqual(jsonWebKey.Kid, theoryData.ComparisonJsonWebKey.Kid, context);
+                    var certificateExpected = (theoryData.SecurityKey as X509SecurityKey).Certificate;
+                    var certificateNew = new X509Certificate2(Convert.FromBase64String(jsonWebKey.X5c[0]));
+                    IdentityComparer.AreEqual(certificateNew.Equals(certificateExpected), true, context);
```
[the remark this answers is not in the document]
Comment (lovemaths) [reply test/Microsoft.IdentityModel.Tokens.Tests/JsonWebKeyConverterTest.cs]: @brentschmaltz Added.

**Comment by brentschmaltz** [comment]
:shipit:

**Pull request merged**